SWLOR.Game.Server/Service/BaseService.cs

```diff
@@ -387,7 +388,28 @@ private static void OnModuleHeartbeat()
                     ClearPCBaseByID(pcBase.ID);
                 }
 
-                var players = module.Players.ToList();
+                // Find all land bases that don't currently have control towers
+                pcBases = DataService.Where<PCBase>(x =>
+                {
+                    var baseStructures = DataService.Where<PCBaseStructure>(y => y.PCBaseID == x.ID && (y.BaseStructureID == 1||y.BaseStructureID ==2|| y.BaseStructureID==3));
+                    return x.PCBaseTypeID == 1 && baseStructures.Count > 0;
+                }).ToList();
+
+                foreach (var pcBase in pcBases)
+                {
+                    Area dbArea = DataService.Single<Area>(x => x.Resref == pcBase.AreaResref);
+
+                    if (pcBase.DateInitialPurchase < DateTime.UtcNow.AddHours(-6))
+                    {
+                        playerIDs.Add(new Tuple<Guid, string>(pcBase.PlayerID, dbArea.Name + " (" + pcBase.Sector + ")"));
+                        ClearPCBaseByID(pcBase.ID);
+                    }
+                    else
+                    {
+                        players.FirstOrDefault(x => x.GlobalID == pcBase.PlayerID)?.FloatingText("You must put a control tower down within 6 hours of purchase or lose your land in " + dbArea.Name + " (" + pcBase.Sector + ")");
```

Comment:
FirstOrDefault runs the chance of returning a null (default = null in this case). Can you swap this to .First() or Single() instead? It probably wouldn't happen in this specific instance, but I want to be on the safe side.